ProcessList.c

```diff
@@ -553,13 +554,39 @@ void ProcessList_rebuildPanel(ProcessList* this) {
    bool foundFollowed = false;
 
    for (int i = 0; i < processCount; i++) {
-      Process* p = (Process*) Vector_get(this->processes, i);
-
-      if ( (!p->show)
-         || (this->userId != (uid_t) -1 && (p->st_uid != this->userId))
-         || (incFilter && !(String_contains_i(Process_getCommand(p), incFilter)))
-         || (this->pidMatchList && !Hashtable_get(this->pidMatchList, p->tgid)) )
-         continue;
+       Process* p = (Process*) Vector_get(this->processes, i);
+
+       if ( (!p->show)
+               || (this->userId != (uid_t) -1 && (p->st_uid != this->userId))
+               || (incFilter && !(String_contains_i(Process_getCommand(p), incFilter))) ) {
+           continue;
+       } else {
+           if (this->pidMatchList && !Hashtable_get(this->pidMatchList, p->tgid)) {
+               // iterate over parents and check hashtable for entries
+               pid_t ppid = Process_getParentPid(p);
+               bool preExisting;
+               bool nextPreExisting;
+               Process *ppidProcess = ProcessList_getProcess(this, ppid, &preExisting, LinuxProcess_new);
+               Process *nextProcess;
+               bool found = Hashtable_get(this->pidMatchList, ppidProcess->tgid);
+               while (ppidProcess->tgid != 0 && !found) {
+                   ppid = Process_getParentPid(ppidProcess);
+                   nextProcess = ProcessList_getProcess(this, ppid, &nextPreExisting, LinuxProcess_new);
+                   found = Hashtable_get(this->pidMatchList, nextProcess->tgid);
+                   if (!preExisting) {
+                       free(ppidProcess);
+                   }
+                   preExisting = nextPreExisting;
+                   ppidProcess = nextProcess;
+               }
+               if (ppidProcess->tgid == 0 && !Hashtable_get(this->pidMatchList, ppidProcess->tgid)) {
+                   if (!preExisting) {
+                        free(ppidProcess);
+                   }
+                   continue;
+               }
```

Comment:
There's a `isChildOf` API available for this kind of filtering; please prefer to use the existing code.

Reply:
I was able to use `isChildOf` for simplifying the while condition, however, I didn't find it helpful for removing the while loop because it didn't search it's parent's parent,  or  parent's grandparent, in the way that iterating over the while loop does. It's definitely simpler now than it was.

I considered adding a `Process_isDescendantOf` function, and pair that with `Hashmap_forEach` to replace the loop, but you'd need a process list built out to pass into a `isDescendantOf` function.

If you have ideas on how to simplify it further, I'd be happy to give them a try.